roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -11,69 +11,34 @@
 - name: set_fact replicas
```

Comment:
I suspect there is a bit of a chicken/egg here, but it seems like we should be able to determine the number of replicas  per-router based on the `replicas`/`selector` value for each router, rather than doing it a single time here.

I know there is some concern with using task includes, but I'm wondering if it might simplify the overall logic here, where all of the logic for a given router could be done in a separate task file, or even the logic to drive each individual router could be pushed up to `tasks/main.yml`

defaults/main.yml:
```
# With multiple routers this should probably default to whatever naming scheme is being used for the individual 
# routers, but we would need to provide some type of backward compatibility to avoid breaking users.
registry_volume_claim: 'registry-claim'
 
openshift_hosted_manage_router: True

openshift_hosted_router_edits:
- key: spec.strategy.rollingParams.intervalSeconds
  value: 1
  action: put
- key: spec.strategy.rollingParams.updatePeriodSeconds
  value: 1
  action: put
- key: spec.strategy.activeDeadlineSeconds
  value: 21600
  action: put

# handle using openshift_fact values as defaults if they are not set explictly
openshift_hosted_router_namespace: "{{ openshift.hosted.router.namespace | default('default') }}"
openshift_hosted_router_selector: "{{ openshift.hosted.router.selector | default(omit) }}"
openshift_hosted_router_image: "{{ openshift.hosted.router.registryurl }}"
openshift_hosted_router_certificate: "{{ openshift.hosted.router.certificate | default({}) }}"

```
tasks/main.yml:
```
- name: Retrieve list of openshift nodes matching router selector
  oc_obj:
    state: list
    kind: node
    namespace: "{{ openshift_hosted_router_namespace }}"
    selector: "{{ openshift_hosted_router_selector }}"
  register: router_nodes

- include: router/router.yml
  when: openshift_hosted_manage_router | default(true) | bool
  with_items:
  - openshift_hosted_routers
  loop_control:
    loop_var: router_item  # override variable used from item to router_item.
    label: "{{ router.name }}"  # override the label to avoid printing out the full datastructure for each iteration
  vars:
    openshift_hosted_router_replicas: "{{ openshift.hosted.router.replicas | default(None)
                                                                | get_router_replicas(router_nodes) }}"
    openshift_hosted_routers:
    - name: router
      replicas: "{{ openshift_hosted_router_replicas }}"
      namespace: "{{ openshift_hosted_router_namespace | default('default') }}"
      serviceaccount: "{{ openshift_hosted_router_serviceaccount | default('router') }}"
      # This option is not yet implemented
      # force_subdomain: "{{ openshift.hosted.router.force_subdomain | default(none) }}"
      selector: "{{ openshift_hosted_router_selector }}"
      images: "{{ openshift_hosted_router_image }}"
      edits: "{{ openshift_hosted_router_edits }}"
      stats_port: 1936
      ports:
      - 80:80
      - 443:443
      certificates: "{{ openshift_hosted_router_certificate | default({}) }}"
```

tasks/router.yml:
```
---
- name: Get the certificate contents for router
  copy:
    backup: True
    dest: "/etc/origin/master/{{ item | basename }}"
    src: "{{ item }}"
  with_items: "{{ router_item.certificates | oo_select_keys_from_list(['keyfile', 'certfile', 'cafile']) }}"

- name: Create OpenShift router
  oc_adm_router:
    name: "{{ router_item.name }}"
    replicas: "{{ router_item.replicas }}"
    namespace: "{{ router_item.namespace }}"
    # This option is not yet implemented
    # force_subdomain: "{{ router_item.force_subdomain }}"
    service_account: "{{ router_item.serviceaccount }}"
    selector: "{{ router_item.selector }}"
    images: "{{ router_item.images }}"
    cert_file: "{{ cert_file }}"
    key_file: "{{ key_file }}"
    cacert_file: "{{ ca_file }}"
    edits: "{{ openshift_hosted_router_edits | union(router_item.edits)  }}"
    ports: "{{ router_item.ports }}"
    stats_port: "{{ router_item.stats_port }}"
  register: routerout
  vars:
    cert_file_basename: "{{ router_item.certificate.certfile | default('') | basename }}"
    key_file_basename: "{{ router_item.certificate.keyfile | default('') | basename }}"
    ca_file_basename: "{{ router_item.certificate.cafile | default('') | basename }}"
    cert_file: "{{ '/etc/origin/master/' ~ cert_file_basename if cert_file_basename != '' else omit }}"
    key_file: "{{ '/etc/origin/master/' ~ key_file_basename if key_file_basename != '' else omit }}"
    ca_file: "{{ '/etc/origin/master/' ~ ca_file_basename if ca_file_basename != '' else omit }}"

# This should probably move to module
- name: wait for deploy
  pause:
    seconds: 30
  when: routerout.changed

- name: Ensure router replica count matches desired
  oc_scale:
    kind: dc
    name: "{{router_item.name }}"
    namespace: "{{ router_item.namespace }}"
    replicas: "{{ router_item.replicas }}"
```